subprojects/testfx-core/src/main/java/org/testfx/matcher/control/ComboBoxMatchers.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    @Factory
+    @Unstable(reason = "is missing apidocs")
+    public static <T> Matcher<Node> containsExactlyItems(T... items) {
+        String descriptionText = "contains exactly items " + Arrays.toString(items);
+        return typeSafeMatcher(ComboBox.class, descriptionText, node -> containsExactlyItems(node, items));
+    }
+
+    @Factory
+    @Unstable(reason = "is missing apidocs")
+    public static <T> Matcher<Node> containsItemsInOrder(T... items) {
+        String descriptionText = "contains items in order " + Arrays.toString(items);
+        return typeSafeMatcher(ComboBox.class, descriptionText, node -> containsItemsInOrder(node, items));
+    }
+
+    @Factory
+    @Unstable(reason = "is missing apidocs")
+    public static <T> Matcher<Node> containsExactlyItemsInOrder(T... items) {
+        String descriptionText = "contains exactly items in order " + Arrays.toString(items);
+        return typeSafeMatcher(ComboBox.class, descriptionText, node -> containsExactlyItemsInOrder(node, items));
+    }
+
+    //---------------------------------------------------------------------------------------------
+    // PRIVATE STATIC METHODS.
+    //---------------------------------------------------------------------------------------------
+
+    private static boolean hasItems(ComboBox<?> comboBox,
+                                    int amount) {
+        return comboBox.getItems().size() == amount;
+    }
+
+    private static <T> boolean hasSelection(ComboBox<?> comboBox,
+                                            T selection) {
+        return selection.equals(comboBox.getSelectionModel().getSelectedItem());
+    }
+
+    private static boolean containsItems(ComboBox<?> comboBox,
+                                         Object... items) {
+        return comboBox.getItems().containsAll(Arrays.asList(items));
+    }
+
+    private static boolean containsExactlyItems(ComboBox<?> comboBox,
+                                         Object... items) {
+        return comboBox.getItems().size() == items.length &&
+                comboBox.getItems().containsAll(Arrays.asList(items));
+    }
+
+    private static boolean containsItemsInOrder(ComboBox<?> comboBox,
+                                                Object... items) {
+        int index = 0;
+
+        // find start of matching sub-sequence
+        while (!comboBox.getItems().get(index).equals(items[0]))
+        {
+            index++;
+        }
+
+        // make sure sub-sequence matches
+        for (Object item : items) {
+            if (!comboBox.getItems().get(index).equals(item)) {
+                return false;
+            }
+            index++;
+        }
+        return true;
+    }
+
+    private static boolean containsExactlyItemsInOrder(ComboBox<?> comboBox,
+                                                       Object... items) {
+        int index = 0;
```

Comment:
Code duplication. This is the same as the last code block in `containsItemsInOrder()`. Could be extracted as private helper method.